### PR TITLE
Code export from VoA for Test Flight quests

### PR DESCRIPTION
**Pull request by killerwife**

### CODE REVIEW

File: src/game/Spell.cpp [hunk 1]
```diff
@@ -6481,10 +6481,11 @@ bool Spell::CheckTarget(Unit* target, SpellEffectIndex eff) const
...
         default:                                            // normal case
             // Get GO cast coordinates if original caster -> GO
-            if (target != m_caster)
-                if (WorldObject* caster = GetCastingObject())
-                    if (!target->IsWithinLOSInMap(caster))
-                        return false;
+            if (!IsIgnoreLosSpell(m_spellInfo))
+                if (target != m_caster)
+                    if (WorldObject* caster = GetCastingObject())
+                        if (!target->IsWithinLOSInMap(caster))
```
Comment (Warlockbugs) [inline src/game/Spell.cpp]: This can be merged into a single line.

Comment (Lillecarl) [inline src/game/Spell.cpp]: It's nice when things are readable too, also how would you make one line out of it when there's an assignment in there? It'd require at least 2 lines?

Comment (Warlockbugs) [inline src/game/Spell.cpp]: You are right about assignment, i meant to get rid of a "stair of ifs" into a usual block. This thing justs asks for a mistake to be made at some point. 😄

**Comment by robinsch** [comment]
This doesn't sound right to me at all. 0x4 contains buffs and dispel magic for example.

**Comment by killerwife** [comment]
Afaik this one was confirmed in some patch notes. And I did my own research when scripting BEM in outland, where several spells have this and need LOS ignoring.

**Comment by robinsch** [comment]
Yes but not in 3.x - 26368 4.0.1 dbc change

**Comment by killerwife** [comment]
Attributes have not changed over the years.

**Comment by robinsch** [comment]
I mean just look at the spells with FamilyName != 0 it just feels wrong.

**Comment by killerwife** [comment]
Despite it looking a bit sketchy, I did put some time and effort into it and I have not run into any issues due to this yet. It is quite possible that some of them have it wrongly. I am watching for any LOS problems on daily basis. So far none.

**Comment by AbaddonL4G** [comment]
Mhm we had a lot of LOS problems with spells with this flag (example: 43215)

found no spell related issue so far (ofc just threw a glance at my list with QtSW)

**Comment by cyberium** [comment]
Partialy merged!

**Comment by cyberium** [comment]
I think this can be closed

**Pull request closed without merging**